File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let answer = [];
    let [a,b,c,d] = ["aya","ye","woo","ma"];
    let arr = [a+a,b+b,c+c,d+d];
    for(let i of babbling){
        if(arr.some(val => i.includes(val))){
            i = "x";
        }
        answer.push( i.split(a).join("z").split(b).join("z").split(c).join("z").split(d).join("z").replaceAll("z",""));
    }
    console.log(answer);
    
    return answer.filter(val => val == "").length;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] arr = {"aya","ye","woo","ma"};
        
        
        for(String s : babbling){
            
            String prev = "";
            
            for(int i=0; i<arr.length; i++){
                if(s.startsWith(arr[i]) && !prev.equals(arr[i])){
                    prev = s.substring(0,arr[i].length());
                    s = s.substring(arr[i].length());
                    i = -1;
                    continue;
                }
            }
            
            if(s.equals("")){
                answer++;
            }
        }
        
        return answer;
    }
}